BasicSteps/NestedTestPlan.cs

```diff
@@ -0,0 +1,136 @@
+        { 
+            // this is copied from TestPlanReference.
+            get => filepath;
+            set
+            {
+                filepath = value;
+                filepath.Context = this;
+                try
+                {
+                    var rp = TapSerializer.GetObjectDeserializer(this)?.ReadPath;
+                    if (rp != null) rp = System.IO.Path.GetDirectoryName(rp);
+                    var fp = filepath.Expand(testPlanDir: rp);
+                    
+                    if (currentlyLoaded != fp)
+                    {
+                        LoadTestPlan();
+                        currentlyLoaded = fp;
+                    }
+                }
+                catch { }
+            }
+        }
+
+        public override void Run()
+        {
+            // the inner plan needs to be reloaded when executed.
+            // this is the simplest way to do that.
+            // Note: the performance penalties of this approach should be considered.
+            var xml = plan.SerializeToString();
+            
+            // for log forwarding, if applicable.
+            LogForwarder forwarder = null;
+
+            // forwarding is only supported in this case.
+            if(ForwardLogEvents && OpenTap.Log.Context is ILogContext2 ctx2)
+                forwarder = new LogForwarder(ctx2, LogHeader);
+            
+            // in the new session, redirect logging in all cases.
+            // create new resources and everything else by overlaying component settings.
+            using (Session.Create(SessionOptions.RedirectLogging | SessionOptions.OverlayComponentSettings))
+            {
+                // load the test plan from XML to ensure getting the new resource references. 
+                var plan2 = Utils.DeserializeFromString<TestPlan>(xml);
+                
+                // don't print the summary of the inner test plan run
+                plan2.PrintTestPlanRunSummary = false;
+                
+                // forward the log messages if enabled.
+                if(forwarder != null)
+                    OpenTap.Log.AddListener(forwarder);
+                
+                // override the plans break conditions with the break conditions of this step.
+                // otherwise the nested plan will get the break conditions from engine settings, which would be
+                // inconsistent with how it otherwise works.
+                BreakConditionProperty.SetBreakCondition(plan2, BreakConditionProperty.GetBreakCondition(this));
+                
+                // this is what actually runs the test plan.
+                // since it is running inside a session, it can be aborted.
+                // it has very limited ways to affect the outer test plan.
+                var subRun = plan2.Execute((ForwardResults ? new IResultListener[] {new ResultForwarder(Results)} as IEnumerable<IResultListener> : ResultSettings.Current));
```

Comment:
What happens if `plan2` requires resources held by the parent test plan?

Reply:
That is why it gets serialized and deserialized. If it is e.g a SCPI instrument, it should work fine. If it is a COM port, it wont work, since the resource becomes locked by first use. 

It depends a bit on how the resources are used. For most, I think it will work.